applications/FSIApplication/custom_utilities/partitioned_fsi_utilities.hpp

```diff
@@ -631,6 +616,64 @@ class PartitionedFSIUtilities
         }
     }
 
+    void CalculateTractionFromPressureValues(
+        ModelPart& rModelPart,
+        const Variable<double>& rPressureVariable,
+        const Variable<array_1d<double,3>>& rTractionVariable,
+        const bool SwapTractionSign)
+    {
+        // Update the nodal normals
+        NormalCalculationUtils().CalculateOnSimplex(rModelPart);
+
+        // Set the nodal traction modulus calculation function
+        std::function<double(const double, const array_1d<double,3>)> traction_modulus_func;
+        if (SwapTractionSign)  {
+            traction_modulus_func = [](const double PosPressure, const array_1d<double,3>& rNormal){return - PosPressure / norm_2(rNormal);};
+        } else {
+            traction_modulus_func = [](const double PosPressure, const array_1d<double,3>& rNormal){return PosPressure / norm_2(rNormal);};
+        }
+
+        // Calculate the tractions from the pressure values
+        block_for_each(rModelPart.Nodes(), [&](Node<3>& rNode){
+            const auto& r_normal = rNode.FastGetSolutionStepValue(NORMAL);
```

Comment:
```suggestion
            const array_1d<double,3>& r_normal = rNode.FastGetSolutionStepValue(NORMAL);
```

remember we had issues with this

Reply:
Yes! Thanks for spotting.